src/loader/BlendLoader.py

```diff
@@ -140,22 +223,30 @@ def run(self):
                 # get all entity's names if not
                 else:
                     entities_to_load = getattr(blend_file_data, attr_name)
-                # load entities
+                    
                 for entity_to_load in entities_to_load:
-
-                    # remove the earlier existing resource with same name
+                    # check if an entity with same name already exists, if 
+                    # it exists rename it
+                    last_obj = None
                     if entity_to_load in bpy.data.objects:
-                        bpy.data.objects.remove(
-                            bpy.data.objects[entity_to_load], do_unlink=True)
+                        last_obj = bpy.data.objects[entity_to_load]
+                        last_obj.name = last_obj.name + '.old'
```

Comment:
Blender automatically renames these objects, accordingly. I would not try to do this manually, as it makes it much harder for us to find objects with the same name.

Reply:
done